### PR TITLE
Marketplace: Adds spinner to search component

### PR DESCRIPTION
**Pull request by Rolilink**
#### Screencast

https://user-images.githubusercontent.com/1035546/164336162-f912dc2c-0b22-4382-b176-8841dce2c59f.mov

#### Changes proposed in this Pull Request

* Adds spinner capability to the marketplace search.

#### Testing instructions

* Go to the marketplace `/plugins/`.
* Search something and the search bar should show a spinner status.

Related to #62496

### CODE REVIEW

**Comment by Rolilink** [comment]
Build is failing because of peer dependencies, need to check what to do in that case.

**Comment by Rolilink** [comment]
> Build is failing because of peer dependencies, need to check what to do in that case.

@jsnajdr do you have any input on why this is happening and how to proceed?

File: packages/search/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-## 1.0.1
+## 1.1.1
+
+- Updates @wordpress/components to v19.8.0 to use the new Spinner design.
+
+## 1.1.0
 
 - Add search modes
 - Add "when-typing" and "on-enter" search mode
```
Comment (lsl) [inline packages/search/CHANGELOG.md]:
```suggestion
- Add search modes, "when-typing" and "on-enter".
```

I noticed @jsnajdr suggested this in the previous PR, lets do this now?

**Comment by Rolilink** [comment]
It seems like the fault is of `airbnb-prop-type`,  `react-dates` and `react-with-direction`.
`react` ones:
```
➤ YN0000: calypso@workspace:client provides react@npm:17.0.2 with version 17.0.2, which doesn't satisfy the following requirements:

➤ YN0000: @automattic/search@workspace:packages/search [abd57] → ^17.0.2                             ✓
➤ YN0000: @emotion/react@npm:11.9.0 [2a70d]                    → >=16.8.0                            ✓
➤ YN0000: @emotion/styled@npm:11.8.1 [2a70d]                   → >=16.8.0                            ✓
➤ YN0000: @wordpress/components@npm:19.4.0 [0b94d]             → ^17.0.0                             ✓
➤ YN0000: @wordpress/compose@npm:5.5.0 [2a70d]                 → ^17.0.0                             ✓
➤ YN0000: @wordpress/data@npm:6.7.0 [f6d72]                    → ^17.0.0                             ✓
➤ YN0000: @wordpress/rich-text@npm:5.5.0 [2a70d]               → ^17.0.0                             ✓
➤ YN0000: airbnb-prop-types@npm:2.16.0 [cdbb3]                 → ^0.14 || ^15.0.0 || ^16.0.0-alpha   ✘
➤ YN0000: framer-motion@npm:4.1.17 [2a70d]                     → >=16.8 || ^17.0.0                   ✓
➤ YN0000: react-dates@npm:17.2.0 [2a70d]                       → ^0.14 || ^15.5.4 || ^16.1.1         ✘
➤ YN0000: react-outside-click-handler@npm:1.3.0 [cdbb3]        → ^0.14 || >=15                       ✓
➤ YN0000: react-portal@npm:4.2.1 [cdbb3]                       → ^15.0.0-0 || ^16.0.0-0 || ^17.0.0-0 ✓
➤ YN0000: react-use-gesture@npm:9.1.3 [2a70d]                  → >= 16.8.0                           ✓
➤ YN0000: react-with-direction@npm:1.3.1 [98227]               → ^0.14 || ^15 || ^16                 ✘
➤ YN0000: react-with-styles@npm:3.2.3 [cdbb3]                  → >=0.14                              ✓

➤ YN0000: Note: these requirements start with @automattic/search@workspace:packages/search [abd57]
```
`react-dom` ones:
```
➤ YN0000: calypso@workspace:client provides react-dom@npm:17.0.2 [35689] with version 17.0.2, which doesn't satisfy the following requirements:

➤ YN0000: @automattic/search@workspace:packages/search [abd57] → ^17.0.2                             ✓
➤ YN0000: @wordpress/components@npm:19.4.0 [0b94d]             → ^17.0.0                             ✓
➤ YN0000: framer-motion@npm:4.1.17 [2a70d]                     → >=16.8 || ^17.0.0                   ✓
➤ YN0000: react-dates@npm:17.2.0 [2a70d]                       → ^0.14 || ^15.5.4 || ^16.1.1         ✘
➤ YN0000: react-outside-click-handler@npm:1.3.0 [cdbb3]        → ^0.14 || >=15                       ✓
➤ YN0000: react-portal@npm:4.2.1 [cdbb3]                       → ^15.0.0-0 || ^16.0.0-0 || ^17.0.0-0 ✓
➤ YN0000: react-with-direction@npm:1.3.1 [98227]               → ^0.14 || ^15 || ^16                 ✘
➤ YN0000: react-with-styles@npm:3.2.3 [cdbb3]                  → *                                   ✓

➤ YN0000: Note: these requirements start with @automattic/search@workspace:packages/search [abd57]
```

**Comment by Rolilink** [comment]
We can merge this one when calypso team folks upgrades all the packages and we have access to the new design p1650573017834329/1650568353.939509-slack-C03DY3RJ5 cc: @noahtallen please if you can drop a comment in here when you finish that.

**Comment by noahtallen** [comment]
Current status on this is here: https://github.com/Automattic/wp-calypso/pull/60161. I think it'll be ready to go tomorrow, though I still have a lot of smoke testing to do! The automated tests seem to be mostly passing now, though, which is a great sign.

**Comment by noahtallen** [comment]
You should be able to move forward with this now! Let me know if you have any other issues :)

**Comment by lsl** [comment]
Rebased, @gcsecsey could you check this PR against the `marketplace-jetpack-plugin-search` flag? I'm not sure how this is supposed to work and testing a couple of searches it looks broken. Not sure if my rebase or something else.

**Comment by lsl** [comment]
Lint failure

```
Replace `⏎↹↹doSearch,⏎↹↹searchTerm,⏎↹↹title,⏎↹↹searchTerms,⏎↹↹isSticky,⏎↹↹popularSearchesRef,⏎↹↹isSearching,⏎↹}·=·` with `·doSearch,·searchTerm,·title,·searchTerms,·isSticky,·popularSearchesRef,·isSearching·}·=⏎↹↹` (prettier/prettier)
```
is this asking for spaces instead of tabs?

**Comment by lsl** [comment]
@Rolilink those test failures look real, I think I broke your PR :smile:

**Comment by gcsecsey** [comment]
@lsl @Rolilink I've pushed a small fix to make the search hooks work with or without the `marketplace-jetpack-plugin-search` flag 👍

**Comment by gcsecsey** [comment]
> Lint failure
> 
> ```
> Replace `⏎↹↹doSearch,⏎↹↹searchTerm,⏎↹↹title,⏎↹↹searchTerms,⏎↹↹isSticky,⏎↹↹popularSearchesRef,⏎↹↹isSearching,⏎↹}·=·` with `·doSearch,·searchTerm,·title,·searchTerms,·isSticky,·popularSearchesRef,·isSearching·}·=⏎↹↹` (prettier/prettier)
> ```
> 
> is this asking for spaces instead of tabs?

In this case prettier is prompting this change:
```diff
- const {
- 	doSearch,
- 	searchTerm,
- 	title,
- 	searchTerms,
- 	isSticky,
- 	popularSearchesRef,
- 	isSearching,
- } = props;
	
+ const { doSearch, searchTerm, title, searchTerms, isSticky, popularSearchesRef, isSearching } =
+	props;
```

It's strange that running `yarn lint:js` locally doesn't pick up this issue (at least for me), but opening the file, the vscode-eslint extension does. Maybe we need to update the `lint:js` script to also run a prettier check?

**Comment by Rolilink** [comment]
thanks @gcsecsey ! LGTM getting this one merged

**Review by lupus2k — COMMENTED** [review]
When filtering for a category, the spinner does not show, also happened with `Top free plugins` but could be because plugins are cached.

Besides that, it looks good. Good job, @Rolilink 🎉

**Review by lupus2k — APPROVED** [review]

**Pull request merged**